none of these projects build

I'm on macOS Sonoma 14.0 intel amd64. 
I've installed and validated that flutter can build and deploy a basic app.

Surprisingly, none of these GetStream flutter-samples projects builds at all.
flutter doctor -v output below.

I tried 2 things, to no avail. I tried changing the gradle version to 8.5 to be compatible with the Java 21 that flutter/android studio uses. No luck. I tried deleting the pubspec_overrides.yaml file. Also no luck. 

How is GetStream even a going concern at all -- if their most basic tech demos fail to build?

~~~
jaten@jbook ~/flutter-samples/packages/stream_chat_v1 (develop) $ ls
ls
README.md		assets			lib			pubspec.yaml
analysis_options.yaml	build			macos			pubspec_overrides.yaml
android			ios			pubspec.lock		web
jaten@jbook ~/flutter-samples/packages/stream_chat_v1 (develop) $ flutter run
flutter run
Launching lib/main.dart on SM A055F in debug mode...
You are applying Flutter's main Gradle plugin imperatively using the apply script method, which is deprecated and will be removed in a future release. Migrate to applying Gradle plugins with the declarative plugins block: https://flutter.dev/to/flutter-gradle-plugin-apply


FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Script '/Users/jaten/flutter/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy' line: 1349

* What went wrong:
A problem occurred evaluating root project 'android'.
> A problem occurred configuring project ':app'.
   > Could not create task ':app:copyFlutterAssetsDebug'.
      > Could not create task ':app:mergeDebugAssets'.
         > Cannot use @TaskAction annotation on method IncrementalTask.taskAction$gradle_core() because interface org.gradle.api.tasks.incremental.IncrementalTaskInputs is not a valid parameter to an action method.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
==============================================================================

2: Task failed with an exception.
-----------
* Where:
Script '/Users/jaten/flutter/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy' line: 1349

* What went wrong:
A problem occurred evaluating root project 'android'.
> A problem occurred configuring project ':app'.
   > Could not create task ':app:copyFlutterAssetsDebug'.
      > Could not create task ':app:mergeDebugAssets'.
         > Cannot use @TaskAction annotation on method IncrementalTask.taskAction$gradle_core() because interface org.gradle.api.tasks.incremental.IncrementalTaskInputs is not a valid parameter to an action method.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
==============================================================================

BUILD FAILED in 10m 19s
Running Gradle task 'assembleDebug'...                            620.1s
Error: Gradle task assembleDebug failed with exit code 1
jaten@jbook ~/flutter-samples/packages/stream_chat_v1 (develop) $ cd ..
cd ..
jaten@jbook ~/flutter-samples/packages (develop) $ ls
ls
chatty		imessage	stream_chat_v1
jaten@jbook ~/flutter-samples/packages (develop) $ cd chatty/
cd chatty/
jaten@jbook ~/flutter-samples/packages/chatty (develop) $ ls
ls
LICENSE		android		assets		lib
README.md	art		ios		pubspec.yaml
jaten@jbook ~/flutter-samples/packages/chatty (develop) $ flutter run
flutter run
Build failed due to use of deleted Android v1 embedding.
jaten@jbook ~/flutter-samples/packages/chatty (develop) $ flutter doctor -v
flutter doctor -v
[✓] Flutter (Channel stable, 3.24.4, on macOS 14.0 23A344 darwin-x64, locale en-US)
    • Flutter version 3.24.4 on channel stable at /Users/jaten/flutter
    • Upstream repository ssh://git@github.com/flutter/flutter.git
    • Framework revision 603104015d (2 weeks ago), 2024-10-24 08:01:25 -0700
    • Engine revision db49896cf2
    • Dart version 3.5.4
    • DevTools version 2.37.3

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at /Users/jaten/Library/Android/sdk
    • Platform android-35, build-tools 35.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915915-b509.11)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915915-b509.11)

[✓] VS Code (version unknown)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.100.0
    ✗ Unable to determine VS Code version.

[✓] Connected device (3 available)
    • SM A055F (mobile) • R92X400L8ZX • android-arm64  • Android 14 (API 34)
    • macOS (desktop)   • macos       • darwin-x64     • macOS 14.0 23A344 darwin-x64
    • Chrome (web)      • chrome      • web-javascript • Google Chrome 130.0.6723.117

[✓] Network resources
    • All expected network resources are available.

• No issues found!
jaten@jbook ~/flutter-samples/packages/chatty (develop) $ git remote -v
git remote -v
origin	ssh://git@github.com/GetStream/flutter-samples (fetch)
origin	ssh://git@github.com/GetStream/flutter-samples (push)
jaten@jbook ~/flutter-samples/packages/chatty (develop) $ cd ..
cd ..
jaten@jbook ~/flutter-samples/packages (develop) $ ls
ls
chatty		imessage	stream_chat_v1
jaten@jbook ~/flutter-samples/packages (develop) $ cd stream_chat_v1/
cd stream_chat_v1/
jaten@jbook ~/flutter-samples/packages/stream_chat_v1 (develop) $ ls
ls
README.md		assets			lib			pubspec.yaml
analysis_options.yaml	build			macos			pubspec_overrides.yaml
android			ios			pubspec.lock		web
jaten@jbook ~/flutter-samples/packages/stream_chat_v1 (develop) $ mv pubspec_overrides.yaml ~
mv pubspec_overrides.yaml ~
jaten@jbook ~/flutter-samples/packages/stream_chat_v1 (develop) $ flutter run
flutter run
Launching lib/main.dart on SM A055F in debug mode...
You are applying Flutter's main Gradle plugin imperatively using the apply script method, which is deprecated and will be removed in a future release. Migrate to applying Gradle plugins with the declarative plugins block: https://flutter.dev/to/flutter-gradle-plugin-apply


FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Script '/Users/jaten/flutter/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy' line: 1349

* What went wrong:
A problem occurred evaluating root project 'android'.
> A problem occurred configuring project ':app'.
   > Could not create task ':app:copyFlutterAssetsDebug'.
      > Could not create task ':app:mergeDebugAssets'.
         > Cannot use @TaskAction annotation on method IncrementalTask.taskAction$gradle_core() because interface org.gradle.api.tasks.incremental.IncrementalTaskInputs is not a valid parameter to an action method.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
==============================================================================

2: Task failed with an exception.
-----------
* Where:
Script '/Users/jaten/flutter/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy' line: 1349

* What went wrong:
A problem occurred evaluating root project 'android'.
> A problem occurred configuring project ':app'.
   > Could not create task ':app:copyFlutterAssetsDebug'.
      > Could not create task ':app:mergeDebugAssets'.
         > Cannot use @TaskAction annotation on method IncrementalTask.taskAction$gradle_core() because interface org.gradle.api.tasks.incremental.IncrementalTaskInputs is not a valid parameter to an action method.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
==============================================================================

BUILD FAILED in 1s
Running Gradle task 'assembleDebug'...                           2,127ms
Error: Gradle task assembleDebug failed with exit code 1
jaten@jbook ~/flutter-samples/packages/stream_chat_v1 (develop) $ flutter run
flutter run
Launching lib/main.dart on SM A055F in debug mode...
You are applying Flutter's main Gradle plugin imperatively using the apply script method, which is deprecated and will be removed in a future release. Migrate to applying Gradle plugins with the declarative plugins block: https://flutter.dev/to/flutter-gradle-plugin-apply


FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Script '/Users/jaten/flutter/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy' line: 1349

* What went wrong:
A problem occurred evaluating root project 'android'.
> A problem occurred configuring project ':app'.
   > Could not create task ':app:copyFlutterAssetsDebug'.
      > Could not create task ':app:mergeDebugAssets'.
         > Cannot use @TaskAction annotation on method IncrementalTask.taskAction$gradle_core() because interface org.gradle.api.tasks.incremental.IncrementalTaskInputs is not a valid parameter to an action method.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
==============================================================================

2: Task failed with an exception.
-----------
* Where:
Script '/Users/jaten/flutter/packages/flutter_tools/gradle/src/main/groovy/flutter.groovy' line: 1349

* What went wrong:
A problem occurred evaluating root project 'android'.
> A problem occurred configuring project ':app'.
   > Could not create task ':app:copyFlutterAssetsDebug'.
      > Could not create task ':app:mergeDebugAssets'.
         > Cannot use @TaskAction annotation on method IncrementalTask.taskAction$gradle_core() because interface org.gradle.api.tasks.incremental.IncrementalTaskInputs is not a valid parameter to an action method.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
==============================================================================

BUILD FAILED in 1s
Running Gradle task 'assembleDebug'...                           1,854ms
Error: Gradle task assembleDebug failed with exit code 1
jaten@jbook ~/flutter-samples/packages/stream_chat_v1 (develop) $ git diff
git diff
diff --git a/packages/imessage/android/build.gradle b/packages/imessage/android/build.gradle
index eec07a3..b1f71e1 100644
--- a/packages/imessage/android/build.gradle
+++ b/packages/imessage/android/build.gradle
@@ -26,6 +26,6 @@ subprojects {
     project.evaluationDependsOn(':app')
 }
 
-task clean(type: Delete) {
+tasks.register("clean", Delete) {
     delete rootProject.buildDir
 }
diff --git a/packages/imessage/android/gradle/wrapper/gradle-wrapper.properties b/packages/imessage/android/gradle/wrapper/gradle-wrapper.properties
index bc6a58a..4cf0c84 100644
--- a/packages/imessage/android/gradle/wrapper/gradle-wrapper.properties
+++ b/packages/imessage/android/gradle/wrapper/gradle-wrapper.properties
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.7-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-all.zip
diff --git a/packages/stream_chat_v1/android/gradle/wrapper/gradle-wrapper.properties b/packages/stream_chat_v1/android/gradle/wrapper/gradle-wrapper.properties
index 75b9065..5fe127a 100644
--- a/packages/stream_chat_v1/android/gradle/wrapper/gradle-wrapper.properties
: 
+++ b/packages/stream_chat_v1/android/gradle/wrapper/gradle-wrapper.properties
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.4-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.5-all.zip
diff --git a/packages/stream_chat_v1/pubspec_overrides.yaml b/packages/stream_chat_v1/pubspec_overrides.yaml
deleted file mode 100644
index 3ae03d0..0000000
--- a/packages/stream_chat_v1/pubspec_overrides.yaml
+++ /dev/null
@@ -1,26 +0,0 @@
-dependency_overrides:
-  stream_chat:
-    git:
-      url: https://github.com/GetStream/stream-chat-flutter.git
-      ref: master
-      path: packages/stream_chat
-  stream_chat_flutter_core:
-    git:
-      url: https://github.com/GetStream/stream-chat-flutter.git
-      ref: master
-      path: packages/stream_chat_flutter_core
-  stream_chat_flutter:
-    git:
-      url: https://github.com/GetStream/stream-chat-flutter.git
-      ref: master
: 
-      path: packages/stream_chat_flutter
-  stream_chat_persistence:
-    git:
-      url: https://github.com/GetStream/stream-chat-flutter.git
-      ref: master
-      path: packages/stream_chat_persistence
-  stream_chat_localizations:
-    git:
-      url: https://github.com/GetStream/stream-chat-flutter.git
-      ref: master
-      path: packages/stream_chat_localizations
\ No newline at end of file
(END) 
(END)q
>jaten@jbook ~/flutter-samples/packages/stream_chat_v1 (develop) $ 
jaten@jbook ~/flutter-samples/packages/stream_chat_v1 (develop) $

jaten@jbook ~/flutter-samples (develop) $ git log
git log
commit 72538f188477c6218f75fc3df306bcb006d3a97a (HEAD -> develop, origin/develop, origin/HEAD)
Merge: 016d7ae d77ff6a
Author: Deven Joshi <deven9852@gmail.com>
Date:   Wed Aug 28 13:25:31 2024 +0200

    Merge pull request #100 from GetStream/fix/update-sample

    fix: Updates stream chat for the project and brings it up to date

commit d77ff6aaabe399ee45a1e32a6f8fcf267f01a612
Author: Deven Joshi <deven9852@gmail.com>
Date:   Tue Aug 20 13:59:26 2024 +0200

    Updates stream chat for the project and brings it up to date

commit 016d7ae4eff22e61cbad97c957d62b7164db2048
Merge: 6ad297e 31e4a0a
Author: Sahil Kumar <xdsahil@gmail.com>
Date:   Tue Aug 22 19:15:21 2023 +0530

    Merge pull request #98 from GetStream/chore/update-project

    chore: update sample app.

commit 31e4a0aafdec35095510866b2317f29d092d66ac

~~~

--- a/.gitignore
+++ b/.gitignore
@@ -35,6 +35,7 @@ gradlew.bat
 gradle-wrapper.jar
 .flutter-plugins-dependencies
 *.iml
+**/android/app/.cxx/
 
 generated_plugin_registrant.dart
 GeneratedPluginRegistrant.h

--- a/packages/imessage/android/app/build.gradle
+++ b/packages/imessage/android/app/build.gradle
@@ -1,14 +1,15 @@
+plugins {
+    id "com.android.application"
+    id "kotlin-android"
+    id "dev.flutter.flutter-gradle-plugin"
+}
+
 def localProperties = new Properties()
 def localPropertiesFile = rootProject.file('local.properties')
 if (localPropertiesFile.exists()) {
     localPropertiesFile.withReader('UTF-8') { reader ->
         localProperties.load(reader)
     }
-}
-
-def flutterRoot = localProperties.getProperty('flutter.sdk')
-if (flutterRoot == null) {
-    throw new GradleException("Flutter SDK not found. Define location with flutter.sdk in the local.properties file.")
 }
 
 def flutterVersionCode = localProperties.getProperty('flutter.versionCode')
@@ -21,12 +22,17 @@ if (flutterVersionName == null) {
     flutterVersionName = '1.0'
 }
 
-apply plugin: 'com.android.application'
-apply plugin: 'kotlin-android'
-apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
-
 android {
-    compileSdkVersion 30
+    namespace "com.example.imessage"
+    compileSdkVersion 35
+
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
+    }
+    kotlinOptions {
+        jvmTarget = '17'
+    }
 
     sourceSets {
         main.java.srcDirs += 'src/main/kotlin'
@@ -36,7 +42,7 @@ android {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
         applicationId "com.example.imessage"
         minSdkVersion 21
-        targetSdkVersion 30
+        targetSdkVersion 35
         versionCode flutterVersionCode.toInteger()
         versionName flutterVersionName
     }
@@ -52,8 +58,4 @@ android {
 
 flutter {
     source '../..'
-}
-
-dependencies {
-    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
 }

--- a/packages/imessage/android/app/src/main/AndroidManifest.xml
+++ b/packages/imessage/android/app/src/main/AndroidManifest.xml
@@ -1,5 +1,4 @@
-<manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.example.imessage">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
    <application
         android:label="imessage"
         android:icon="@mipmap/ic_launcher">
@@ -9,6 +8,7 @@
             android:theme="@style/LaunchTheme"
             android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
             android:hardwareAccelerated="true"
+            android:exported="true"
             android:windowSoftInputMode="adjustResize">
             <!-- Specifies an Android theme to apply to this Activity as soon as
                  the Android process has started. This theme is visible to the user

--- a/packages/imessage/android/build.gradle
+++ b/packages/imessage/android/build.gradle
@@ -1,16 +1,3 @@
-buildscript {
-    ext.kotlin_version = '1.5.20'
-    repositories {
-        google()
-        jcenter()
-    }
-
-    dependencies {
-        classpath 'com.android.tools.build:gradle:4.1.0'
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
-    }
-}
-
 allprojects {
     repositories {
         google()
@@ -26,6 +13,6 @@ subprojects {
     project.evaluationDependsOn(':app')
 }
 
-task clean(type: Delete) {
-    delete rootProject.buildDir
+tasks.register("clean", Delete) {
+    delete rootProject.layout.buildDirectory
 }

--- a/packages/imessage/android/gradle.properties
+++ b/packages/imessage/android/gradle.properties
@@ -1,4 +1,3 @@
 org.gradle.jvmargs=-Xmx1536M
 android.useAndroidX=true
 android.enableJetifier=true
-android.enableR8=true

--- a/packages/imessage/android/gradle/wrapper/gradle-wrapper.properties
+++ b/packages/imessage/android/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
-#Fri Jun 23 08:50:38 CEST 2017
+#Fri Feb 21 14:17:43 CET 2025
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.10.2-all.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.7-all.zip

--- a/packages/imessage/android/settings.gradle
+++ b/packages/imessage/android/settings.gradle
@@ -1,11 +1,25 @@
-include ':app'
+pluginManagement {
+    def flutterSdkPath = {
+        def properties = new Properties()
+        file("local.properties").withInputStream { properties.load(it) }
+        def flutterSdkPath = properties.getProperty("flutter.sdk")
+        assert flutterSdkPath != null, "flutter.sdk not set in local.properties"
+        return flutterSdkPath
+    }()
 
-def localPropertiesFile = new File(rootProject.projectDir, "local.properties")
-def properties = new Properties()
+    includeBuild("$flutterSdkPath/packages/flutter_tools/gradle")
 
-assert localPropertiesFile.exists()
-localPropertiesFile.withReader("UTF-8") { reader -> properties.load(reader) }
+    repositories {
+        google()
+        mavenCentral()
+        gradlePluginPortal()
+    }
+}
 
-def flutterSdkPath = properties.getProperty("flutter.sdk")
-assert flutterSdkPath != null, "flutter.sdk not set in local.properties"
-apply from: "$flutterSdkPath/packages/flutter_tools/gradle/app_plugin_loader.gradle"
+plugins {
+    id "dev.flutter.flutter-plugin-loader" version "1.0.0"
+    id "com.android.application" version "8.8.1" apply false
+    id "org.jetbrains.kotlin.android" version "1.9.10" apply false
+}
+
+include ":app"

--- a/packages/imessage/ios/Runner.xcodeproj/project.pbxproj
+++ b/packages/imessage/ios/Runner.xcodeproj/project.pbxproj
@@ -3,17 +3,17 @@
 	archiveVersion = 1;
 	classes = {
 	};
-	objectVersion = 46;
+	objectVersion = 54;
 	objects = {
 
 /* Begin PBXBuildFile section */
 		1498D2341E8E89220040F4C2 /* GeneratedPluginRegistrant.m in Sources */ = {isa = PBXBuildFile; fileRef = 1498D2331E8E89220040F4C2 /* GeneratedPluginRegistrant.m */; };
 		3B3967161E833CAA004F5970 /* AppFrameworkInfo.plist in Resources */ = {isa = PBXBuildFile; fileRef = 3B3967151E833CAA004F5970 /* AppFrameworkInfo.plist */; };
+		672034BE764343416BBF054C /* Pods_Runner.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 68C54947B67A97B17A212D7D /* Pods_Runner.framework */; };
 		74858FAF1ED2DC5600515810 /* AppDelegate.swift in Sources */ = {isa = PBXBuildFile; fileRef = 74858FAE1ED2DC5600515810 /* AppDelegate.swift */; };
 		97C146FC1CF9000F007C117D /* Main.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FA1CF9000F007C117D /* Main.storyboard */; };
 		97C146FE1CF9000F007C117D /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FD1CF9000F007C117D /* Assets.xcassets */; };
 		97C147011CF9000F007C117D /* LaunchScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FF1CF9000F007C117D /* LaunchScreen.storyboard */; };
-		AD80798B75BEF39A28B8324C /* Pods_Runner.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = D0EA12FA64E96796BC730E03 /* Pods_Runner.framework */; };
 /* End PBXBuildFile section */
 
 /* Begin PBXCopyFilesBuildPhase section */
@@ -30,15 +30,14 @@
 /* End PBXCopyFilesBuildPhase section */
 
 /* Begin PBXFileReference section */
-		10AA3A20A8E8F51016843518 /* Pods-Runner.release.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-Runner.release.xcconfig"; path = "Target Support Files/Pods-Runner/Pods-Runner.release.xcconfig"; sourceTree = "<group>"; };
 		1498D2321E8E86230040F4C2 /* GeneratedPluginRegistrant.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = GeneratedPluginRegistrant.h; sourceTree = "<group>"; };
 		1498D2331E8E89220040F4C2 /* GeneratedPluginRegistrant.m */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.objc; path = GeneratedPluginRegistrant.m; sourceTree = "<group>"; };
-		212173E324AE0FED45F2866A /* Pods-Runner.profile.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-Runner.profile.xcconfig"; path = "Target Support Files/Pods-Runner/Pods-Runner.profile.xcconfig"; sourceTree = "<group>"; };
 		3B3967151E833CAA004F5970 /* AppFrameworkInfo.plist */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.plist.xml; name = AppFrameworkInfo.plist; path = Flutter/AppFrameworkInfo.plist; sourceTree = "<group>"; };
+		68C54947B67A97B17A212D7D /* Pods_Runner.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; includeInIndex = 0; path = Pods_Runner.framework; sourceTree = BUILT_PRODUCTS_DIR; };
 		74858FAD1ED2DC5600515810 /* Runner-Bridging-Header.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = "Runner-Bridging-Header.h"; sourceTree = "<group>"; };
 		74858FAE1ED2DC5600515810 /* AppDelegate.swift */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.swift; path = AppDelegate.swift; sourceTree = "<group>"; };
+		7804032A8DB1A80610037311 /* Pods-Runner.debug.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-Runner.debug.xcconfig"; path = "Target Support Files/Pods-Runner/Pods-Runner.debug.xcconfig"; sourceTree = "<group>"; };
 		7AFA3C8E1D35360C0083082E /* Release.xcconfig */ = {isa = PBXFileReference; lastKnownFileType = text.xcconfig; name = Release.xcconfig; path = Flutter/Release.xcconfig; sourceTree = "<group>"; };
-		87BC4F0CB1D55BC3C77185AD /* Pods-Runner.debug.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-Runner.debug.xcconfig"; path = "Target Support Files/Pods-Runner/Pods-Runner.debug.xcconfig"; sourceTree = "<group>"; };
 		9740EEB21CF90195004384FC /* Debug.xcconfig */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.xcconfig; name = Debug.xcconfig; path = Flutter/Debug.xcconfig; sourceTree = "<group>"; };
 		9740EEB31CF90195004384FC /* Generated.xcconfig */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = text.xcconfig; name = Generated.xcconfig; path = Flutter/Generated.xcconfig; sourceTree = "<group>"; };
 		97C146EE1CF9000F007C117D /* Runner.app */ = {isa = PBXFileReference; explicitFileType = wrapper.application; includeInIndex = 0; path = Runner.app; sourceTree = BUILT_PRODUCTS_DIR; };
@@ -46,7 +45,8 @@
 		97C146FD1CF9000F007C117D /* Assets.xcassets */ = {isa = PBXFileReference; lastKnownFileType = folder.assetcatalog; path = Assets.xcassets; sourceTree = "<group>"; };
 		97C147001CF9000F007C117D /* Base */ = {isa = PBXFileReference; lastKnownFileType = file.storyboard; name = Base; path = Base.lproj/LaunchScreen.storyboard; sourceTree = "<group>"; };
 		97C147021CF9000F007C117D /* Info.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist.xml; path = Info.plist; sourceTree = "<group>"; };
-		D0EA12FA64E96796BC730E03 /* Pods_Runner.framework */ = {isa = PBXFileReference; explicitFileType = wrapper.framework; includeInIndex = 0; path = Pods_Runner.framework; sourceTree = BUILT_PRODUCTS_DIR; };
+		9BAE28C557F7C8298AEBE685 /* Pods-Runner.release.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-Runner.release.xcconfig"; path = "Target Support Files/Pods-Runner/Pods-Runner.release.xcconfig"; sourceTree = "<group>"; };
+		C6E1DD8F5DA1BBEF18BEDBCA /* Pods-Runner.profile.xcconfig */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.xcconfig; name = "Pods-Runner.profile.xcconfig"; path = "Target Support Files/Pods-Runner/Pods-Runner.profile.xcconfig"; sourceTree = "<group>"; };
 /* End PBXFileReference section */
 
 /* Begin PBXFrameworksBuildPhase section */
@@ -54,27 +54,27 @@
 			isa = PBXFrameworksBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
-				AD80798B75BEF39A28B8324C /* Pods_Runner.framework in Frameworks */,
+				672034BE764343416BBF054C /* Pods_Runner.framework in Frameworks */,
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 		};
 /* End PBXFrameworksBuildPhase section */
 
 /* Begin PBXGroup section */
-		3F98BA59D9BB09C5816FF126 /* Frameworks */ = {
+		7B31D8C32330F8D3BDF8768B /* Frameworks */ = {
 			isa = PBXGroup;
 			children = (
-				D0EA12FA64E96796BC730E03 /* Pods_Runner.framework */,
+				68C54947B67A97B17A212D7D /* Pods_Runner.framework */,
 			);
 			name = Frameworks;
 			sourceTree = "<group>";
 		};
-		7231B8C27F5ACE1DDE056446 /* Pods */ = {
+		860C8405C4C1076EEE6EC118 /* Pods */ = {
 			isa = PBXGroup;
 			children = (
-				87BC4F0CB1D55BC3C77185AD /* Pods-Runner.debug.xcconfig */,
-				10AA3A20A8E8F51016843518 /* Pods-Runner.release.xcconfig */,
-				212173E324AE0FED45F2866A /* Pods-Runner.profile.xcconfig */,
+				7804032A8DB1A80610037311 /* Pods-Runner.debug.xcconfig */,
+				9BAE28C557F7C8298AEBE685 /* Pods-Runner.release.xcconfig */,
+				C6E1DD8F5DA1BBEF18BEDBCA /* Pods-Runner.profile.xcconfig */,
 			);
 			name = Pods;
 			path = Pods;
@@ -97,8 +97,8 @@
 				9740EEB11CF90186004384FC /* Flutter */,
 				97C146F01CF9000F007C117D /* Runner */,
 				97C146EF1CF9000F007C117D /* Products */,
-				7231B8C27F5ACE1DDE056446 /* Pods */,
-				3F98BA59D9BB09C5816FF126 /* Frameworks */,
+				860C8405C4C1076EEE6EC118 /* Pods */,
+				7B31D8C32330F8D3BDF8768B /* Frameworks */,
 			);
 			sourceTree = "<group>";
 		};
@@ -132,14 +132,14 @@
 			isa = PBXNativeTarget;
 			buildConfigurationList = 97C147051CF9000F007C117D /* Build configuration list for PBXNativeTarget "Runner" */;
 			buildPhases = (
-				CAA82304056351F698F2FE92 /* [CP] Check Pods Manifest.lock */,
+				70BE71C1BA167AD723F04822 /* [CP] Check Pods Manifest.lock */,
 				9740EEB61CF901F6004384FC /* Run Script */,
 				97C146EA1CF9000F007C117D /* Sources */,
 				97C146EB1CF9000F007C117D /* Frameworks */,
 				97C146EC1CF9000F007C117D /* Resources */,
 				9705A1C41CF9048500538489 /* Embed Frameworks */,
 				3B06AD1E1E4923F5004D2608 /* Thin Binary */,
-				5C9569EFDB9DD32B2BAFBEBB /* [CP] Embed Pods Frameworks */,
+				3DA49115DFBB33B00233918E /* [CP] Embed Pods Frameworks */,
 			);
 			buildRules = (
 			);
@@ -156,7 +156,7 @@
 		97C146E61CF9000F007C117D /* Project object */ = {
 			isa = PBXProject;
 			attributes = {
-				LastUpgradeCheck = 1020;
+				LastUpgradeCheck = 1510;
 				ORGANIZATIONNAME = "";
 				TargetAttributes = {
 					97C146ED1CF9000F007C117D = {
@@ -200,10 +200,12 @@
 /* Begin PBXShellScriptBuildPhase section */
 		3B06AD1E1E4923F5004D2608 /* Thin Binary */ = {
 			isa = PBXShellScriptBuildPhase;
+			alwaysOutOfDate = 1;
 			buildActionMask = 2147483647;
 			files = (
 			);
 			inputPaths = (
+				"${TARGET_BUILD_DIR}/${INFOPLIST_PATH}",
 			);
 			name = "Thin Binary";
 			outputPaths = (
@@ -212,7 +214,7 @@
 			shellPath = /bin/sh;
 			shellScript = "/bin/sh \"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh\" embed_and_thin";
 		};
-		5C9569EFDB9DD32B2BAFBEBB /* [CP] Embed Pods Frameworks */ = {
+		3DA49115DFBB33B00233918E /* [CP] Embed Pods Frameworks */ = {
 			isa = PBXShellScriptBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
@@ -229,21 +231,7 @@
 			shellScript = "\"${PODS_ROOT}/Target Support Files/Pods-Runner/Pods-Runner-frameworks.sh\"\n";
 			showEnvVarsInLog = 0;
 		};
-		9740EEB61CF901F6004384FC /* Run Script */ = {
-			isa = PBXShellScriptBuildPhase;
-			buildActionMask = 2147483647;
-			files = (
-			);
-			inputPaths = (
-			);
-			name = "Run Script";
-			outputPaths = (
-			);
-			runOnlyForDeploymentPostprocessing = 0;
-			shellPath = /bin/sh;
-			shellScript = "/bin/sh \"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh\" build";
-		};
-		CAA82304056351F698F2FE92 /* [CP] Check Pods Manifest.lock */ = {
+		70BE71C1BA167AD723F04822 /* [CP] Check Pods Manifest.lock */ = {
 			isa = PBXShellScriptBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
@@ -264,6 +252,21 @@
 			shellPath = /bin/sh;
 			shellScript = "diff \"${PODS_PODFILE_DIR_PATH}/Podfile.lock\" \"${PODS_ROOT}/Manifest.lock\" > /dev/null\nif [ $? != 0 ] ; then\n    # print error to STDERR\n    echo \"error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.\" >&2\n    exit 1\nfi\n# This output is used by Xcode 'outputs' to avoid re-running this script phase.\necho \"SUCCESS\" > \"${SCRIPT_OUTPUT_FILE_0}\"\n";
 			showEnvVarsInLog = 0;
+		};
+		9740EEB61CF901F6004384FC /* Run Script */ = {
+			isa = PBXShellScriptBuildPhase;
+			alwaysOutOfDate = 1;
+			buildActionMask = 2147483647;
+			files = (
+			);
+			inputPaths = (
+			);
+			name = "Run Script";
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "/bin/sh \"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh\" build";
 		};
 /* End PBXShellScriptBuildPhase section */
 
@@ -340,7 +343,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
 				MTL_ENABLE_DEBUG_INFO = NO;
 				SDKROOT = iphoneos;
 				SUPPORTED_PLATFORMS = iphoneos;
@@ -362,7 +365,10 @@
 					"$(PROJECT_DIR)/Flutter",
 				);
 				INFOPLIST_FILE = Runner/Info.plist;
-				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
+				LD_RUNPATH_SEARCH_PATHS = (
+					"$(inherited)",
+					"@executable_path/Frameworks",
+				);
 				LIBRARY_SEARCH_PATHS = (
 					"$(inherited)",
 					"$(PROJECT_DIR)/Flutter",
@@ -422,7 +428,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
 				MTL_ENABLE_DEBUG_INFO = YES;
 				ONLY_ACTIVE_ARCH = YES;
 				SDKROOT = iphoneos;
@@ -471,7 +477,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 9.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
 				MTL_ENABLE_DEBUG_INFO = NO;
 				SDKROOT = iphoneos;
 				SUPPORTED_PLATFORMS = iphoneos;
@@ -494,7 +500,10 @@
 					"$(PROJECT_DIR)/Flutter",
 				);
 				INFOPLIST_FILE = Runner/Info.plist;
-				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
+				LD_RUNPATH_SEARCH_PATHS = (
+					"$(inherited)",
+					"@executable_path/Frameworks",
+				);
 				LIBRARY_SEARCH_PATHS = (
 					"$(inherited)",
 					"$(PROJECT_DIR)/Flutter",
@@ -521,7 +530,10 @@
 					"$(PROJECT_DIR)/Flutter",
 				);
 				INFOPLIST_FILE = Runner/Info.plist;
-				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
+				LD_RUNPATH_SEARCH_PATHS = (
+					"$(inherited)",
+					"@executable_path/Frameworks",
+				);
 				LIBRARY_SEARCH_PATHS = (
 					"$(inherited)",
 					"$(PROJECT_DIR)/Flutter",

--- a/packages/imessage/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme
+++ b/packages/imessage/ios/Runner.xcodeproj/xcshareddata/xcschemes/Runner.xcscheme
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "1020"
+   LastUpgradeVersion = "1510"
    version = "1.3">
    <BuildAction
       parallelizeBuildables = "YES"
@@ -50,6 +50,7 @@
       ignoresPersistentStateOnLaunch = "NO"
       debugDocumentVersioning = "YES"
       debugServiceExtension = "internal"
+      enableGPUValidationMode = "1"
       allowLocationSimulation = "YES">
       <BuildableProductRunnable
          runnableDebuggingMode = "0">

--- a/packages/imessage/ios/Runner/AppDelegate.swift
+++ b/packages/imessage/ios/Runner/AppDelegate.swift
@@ -1,7 +1,7 @@
 import UIKit
 import Flutter
 
-@UIApplicationMain
+@main
 @objc class AppDelegate: FlutterAppDelegate {
   override func application(
     _ application: UIApplication,

--- a/packages/imessage/ios/Runner/Info.plist
+++ b/packages/imessage/ios/Runner/Info.plist
@@ -41,5 +41,9 @@
 	</array>
 	<key>UIViewControllerBasedStatusBarAppearance</key>
 	<false/>
+	<key>CADisableMinimumFrameDurationOnPhone</key>
+	<true/>
+	<key>UIApplicationSupportsIndirectInputEvents</key>
+	<true/>
 </dict>
 </plist>

--- a/packages/imessage/lib/channel_list_view.dart
+++ b/packages/imessage/lib/channel_list_view.dart
@@ -11,7 +11,9 @@ class ChannelListView extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    channels.removeWhere((channel) => channel.lastMessageAt == null);
+    final filteredChannels = List.from(channels)
+      ..removeWhere((channel) => channel.lastMessageAt == null);
+
     return SliverList(
       delegate: SliverChildBuilderDelegate(
         (
@@ -21,13 +23,13 @@ class ChannelListView extends StatelessWidget {
           return Padding(
             padding: const EdgeInsets.symmetric(vertical: 4.0),
             child: ChannelPreview(
-              channel: channels[index],
+              channel: filteredChannels[index],
               onTap: () {
                 Navigator.push(
                   context,
                   PageRouteBuilder(
                     pageBuilder: (_, __, ___) => StreamChannel(
-                      channel: channels[index],
+                      channel: filteredChannels[index],
                       child: const MessagePage(),
                     ),
                     transitionsBuilder: (
@@ -48,7 +50,7 @@ class ChannelListView extends StatelessWidget {
             ),
           );
         },
-        childCount: channels.length,
+        childCount: filteredChannels.length,
       ),
     );
   }

--- a/packages/imessage/lib/main.dart
+++ b/packages/imessage/lib/main.dart
@@ -42,68 +42,90 @@ class IMessage extends StatelessWidget {
   }
 }
 
-class ChatLoader extends StatelessWidget {
-  ChatLoader({
+class ChatLoader extends StatefulWidget {
+  const ChatLoader({
     Key? key,
   }) : super(key: key);
 
-  final channelListController = ChannelListController();
+  @override
+  State<ChatLoader> createState() => _ChatLoaderState();
+}
+
+class _ChatLoaderState extends State<ChatLoader> {
+  StreamChannelListController? _channelListController;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    if (_channelListController
+        case final StreamChannelListController controller) {
+      controller.dispose();
+    }
+
+    final streamChat = StreamChatCore.of(context);
+
+    _channelListController = StreamChannelListController(
+      client: streamChat.client,
+      filter: Filter.and([
+        Filter.in_('members', [streamChat.currentUser!.id]),
+        Filter.equal('type', 'messaging'),
+      ]),
+      channelStateSort: const [SortOption('last_message_at')],
+      limit: 20,
+    )..doInitialLoad();
+  }
+
+  @override
+  void dispose() {
+    _channelListController?.dispose();
+    super.dispose();
+  }
 
   @override
   Widget build(BuildContext context) {
-    final user = StreamChatCore.of(context).currentUser!;
+    final channelListController = _channelListController!;
+
     return CupertinoPageScaffold(
-      child: ChannelsBloc(
-        child: ChannelListCore(
-          channelListController: channelListController,
-          filter: Filter.and([
-            Filter.in_('members', [user.id]),
-            Filter.equal('type', 'messaging'),
-          ]),
-          sort: const [SortOption('last_message_at')],
-          limit: 20,
-          emptyBuilder: (BuildContext context) {
-            return const Center(
-              child: Text('Looks like you are not in any channels'),
-            );
-          },
-          loadingBuilder: (BuildContext context) {
-            return const Center(
+      child: PagedValueListenableBuilder<int, Channel>(
+        valueListenable: channelListController,
+        builder: (context, value, child) {
+          return value.when(
+            (channels, nextPageKey, error) => LazyLoadScrollView(
+              onEndOfPage: () async {
+                if (nextPageKey != null) {
+                  channelListController.loadMore(nextPageKey);
+                }
+              },
+              child: channels.isEmpty
+                  ? const Center(
+                      child: Text('Looks like you are not in any channels'),
+                    )
+                  : CustomScrollView(
+                      slivers: [
+                        CupertinoSliverRefreshControl(onRefresh: () async {
+                          return channelListController.refresh();
+                        }),
+                        const ChannelPageAppBar(),
+                        SliverPadding(
+                          sliver: ChannelListView(channels: channels),
+                          padding: const EdgeInsets.only(top: 16),
+                        )
+                      ],
+                    ),
+            ),
+            loading: () => const Center(
               child: SizedBox(
                 height: 100.0,
                 width: 100.0,
                 child: CupertinoActivityIndicator(),
               ),
-            );
-          },
-          errorBuilder: (BuildContext context, dynamic error) {
-            return const Center(
+            ),
+            error: (e) => const Center(
               child: Text(
                   'Oh no, something went wrong. Please check your config.'),
-            );
-          },
-          listBuilder: (
-            BuildContext context,
-            List<Channel> channels,
-          ) =>
-              LazyLoadScrollView(
-            onEndOfPage: () async {
-              return channelListController.paginateData!();
-            },
-            child: CustomScrollView(
-              slivers: [
-                CupertinoSliverRefreshControl(onRefresh: () async {
-                  return channelListController.loadData!();
-                }),
-                const ChannelPageAppBar(),
-                SliverPadding(
-                  sliver: ChannelListView(channels: channels),
-                  padding: const EdgeInsets.only(top: 16),
-                )
-              ],
             ),
-          ),
-        ),
+          );
+        },
       ),
     );
   }

--- a/packages/imessage/pubspec.yaml
+++ b/packages/imessage/pubspec.yaml
@@ -18,25 +18,26 @@ publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 version: 1.0.0+1
 
 environment:
-  sdk: '>=2.12.0 <3.0.0'
+  sdk: ^3.6.0
+  flutter: ">=3.27.0"
 
 dependencies:
   flutter:
     sdk: flutter
-  intl: ^0.17.0
-  stream_chat_flutter_core: ^3.0.0
+  intl: ^0.20.2
+  stream_chat_flutter_core: ^9.4.0
   animations: ^2.0.1
   collection: ^1.15.0
   cached_network_image: ^3.1.0
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^1.0.3
-  image_picker: ^0.8.4+4
+  image_picker: ^1.1.2
 
 dev_dependencies:
   flutter_test:
     sdk: flutter
-  flutter_lints: ^1.0.0
+  flutter_lints: ^5.0.0
 
 # For information on the generic Dart part of this file, see the
 # following page: https://dart.dev/tools/pub/pubspec

